pkg/api/domains/domains_test.go

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+package domains
+
+import (
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	v8weblogic "github.com/verrazzano/verrazzano-crd-generator/pkg/apis/weblogic/v8"
+	"github.com/verrazzano/verrazzano-operator/pkg/controller"
+	"github.com/verrazzano/verrazzano-operator/pkg/testutil"
+	"github.com/verrazzano/verrazzano-operator/pkg/testutilcontroller"
+	"github.com/verrazzano/verrazzano-operator/pkg/types"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// TestInit ensures that a set of domains can be obtained from the provided set of listers.
+func TestInit(t *testing.T) {
+	// GIVEN an array of clusters and a model binding pair
+	Domains = nil
```

Comment:
I know we have global vars like Domain used in many of these other api subpackages, but this is bad.  At the very least we should have a setter rather than setting the var from a different go file.  Also, lower case Domain to make it package private if you could.